scripts/core/world/world.js

```diff
@@ -66,22 +107,31 @@ define([
                 }
 
                 var agent = Agent.createNewAgent(generator, location);
-                agent.setID(_.size(agents));
+                agent.setID(_.size(agentsByID));
+                agent.setWorld(myself);
+
+                agentsByID[agent.getID()] = agent;
+                myself.updateAgentPerLocation(agent, agent.getLocation());
 
-                agents.push(agent);
                 return agent;
             }
 
             this.getAgent = function(id) {
-                return agents[id];
+                return agentsByID[id];
             }
 
             this.getAgentQuantity = function() {
-                return _.size(agents);
+                return _.size(agentsByID);
             }
 
             this.getAllAgents = function() {
-                return agents;
+                return agentsByID;
+            }
+
+            this.getAgentsAt = function(location) {
```

Comment:
Here is for you @jacklehamster , You can give it a location object and it will return all the Agent at those coordinates.